source/upgrade.txt

```diff
@@ -109,13 +109,102 @@ Version 5.0 Breaking Changes
 - This driver version removes the
   ``org.mongodb.scala.ObservableImplicits.ToSingleObservableVoid`` implicit
   class. This means the ``org.reactivestreams.Publisher[Void]`` type no longer
-  converts automatically to ``org.mongodb.scala.SingleObservable[Void]``. The also
-  API exposes ``org.mongodb.scala.Observable[Unit]`` instead of
+  converts automatically to ``org.mongodb.scala.SingleObservable[Void]``. The
+  API also exposes ``org.mongodb.scala.Observable[Unit]`` instead of
   ``org.mongodb.scala.Observable[Void]``. 
   
   .. After the 5.0 Scala API docs are released, this line will be uncommented. 
     For more information, see the `Observable trait documentation <https://mongodb.github.io/mongo-java-driver/5.0/apidocs/mongo-scala-driver/org/mongodb/scala/Observable.html>`__.
 
+- This driver version removes the ``Parameterizable`` interface. Instead of
+  implementing this interface on a custom ``Codec`` type,
+  override the ``CodecProvider.get()`` method on the
+  codec's ``CodecProvider`` if the codec is intended for a parameterized
+  type.
+
+- This driver version removes the ``isSlaveOk()`` method from the
+  ``ReadPreference`` and ``TaggableReadPreference`` classes. To check whether a read preference allows
+  reading from a secondary member of a replica set, use the ``isSecondaryOk()`` methods from
+  these classes instead.
+
+- This driver version removes the ``DBCollection.getStats()`` and ``DBCollection.isCapped()``
+  helper methods for the ``collStats`` command. Instead of these methods, you can use the
+  ``$collStats`` aggregation pipeline stage. For an example of how to use this pipeline
+  stage, see :manual:`What's New <java-deprecations-4.11>` for v4.11 of the {+driver-short+}.
+
+- This driver version removes the ``IterableCodec`` and ``MapCodec`` types.
+  Use ``MapCodecProvider`` instead of ``MapCodec`` and ``CollectionCodecProvider``
+  instead of ``IterableCodec``.
+
+- This driver version removes the ``sharded`` and ``nonAtomic`` options from the
+  ``MapReducePublisher`` and ``MapReduceIterable`` classes.
+
+- This driver version removes the following methods for use with ``geoHaystack`` indexes:
+
+  - ``Indexes.geoHaystack()``
+  - ``IndexOptions.getBucketSize()``
+  - ``IndexOptions.bucketSize()``
+  
+  Instead, you can use the ``$geoNear`` aggregation pipeline stage or a geospatial
+  query operator on a 2d index. For more information, see the
+  :manual:`Geospatial Queries page </geospatial-queries>` in the {+mdb-server+} manual.
+
+- This driver version removes the ``oplogReplay`` option from find operations. This
+  includes the following methods:
+
+  - ``DBCursor.oplogReplay()``
+  - ``DBCollectionFindOptions.isOplogReplay()``
+  - ``DBCollectionFindOptions.oplogReplay()``
+  - ``FindPublisher.oplogReplay()``
+  - ``FindIterable.oplogReplay()``
+
+- This driver version removes the following ``Exception`` constructors:
```

Comment:
In many cases where the DBX team removed constructors from a class, they left a single, all-encompassing constructor. To avoid these laundry lists, it might be cleaner to just say that the classes only contain a single constructor. Let me know.

Reply:
I understand that it would be cleaner to write it that way, but writing out each constructor makes this breaking issue more searchable.

Reply:
That's a really good point, and one I hadn't considered.